docs/2022-07-18-JSON-RPC-Server-API.md

```diff
@@ -388,6 +388,127 @@ Same as for execute
 }
 ```
 
+## Simplify Implication
+
+### Request:
+
+```json
+{
+  "jsonrpc": "2.0",
+  "id": 1,
+  "method": "simplify-implication",
+  "params": {
+    "antecedent": {"format": "KORE", "version": 1, "term": {}},
+    "consequent": {"format": "KORE", "version": 1, "term": {}},
+    "module": "MODULE-NAME"
+  }
+}
+```
+
+Optional parameters: `module` (main module name)
+
+The request format is shared with the `"implies"` method.
+
+**NOTE**: `"simplify-implication"` currently only has a stub implementation in `kore-rpc`. The real implementation is in `kore-rpc-booster` (see [hs-backend-booster](https://github.com/runtimeverification/hs-backend-booster) repository). The documentation will reside here for consistency.
+
+### Error Response:
+
+Errors in decoding the `antecedent` or `consequent` terms are reported similar as for execute.
+
+### Correct Response:
+
+The endpoint is a lightweight variant fo the `"implies"` endpoint, which checks implication using matching between configuration terms and a lightweight (using K simplifications, rather than encoding to SMT) constraint subsumption.
+
+The implication can be "valid", "invalid" and "unknown". The result is returned in the `"validity"` field. The following results are possible:
+
+#### Implication is **valid**
+
+A constrained substitution as the result, and this is only returned if the implication is valid.
+
+```
+{
+  "jsonrpc": "2.0",
+  "id": 1,
+  "result": {
+    "validity": {"tag": "implication-valid"},
+    "substitution": {"format": "KORE", "version": 1, "term": {}},
+  }
+}
+```
+
+#### Implication **invalid**: terms do not match
+
+Matching between antecedent and consequent configurations failed (different constructors, shared variables, sort error, etc.), constraints has not been subsumption been attempted. No matching substitution is returned.
+
+```
+{
+  "jsonrpc": "2.0",
+  "id": 1,
+  "result": {
+    "validity": {"tag": "implication-valid",
+                 "contents": {"tag": "matching-failed",
+                              "contents": "Shared variables: X:SortWordStack{}"
+                 }},
+  }
+}
+```
```

Comment:
The result here should be "unification-failed", and that's what we should be checking. Eventually, if unification succeeds and matching fails, we want ot return an core problem which may or may not be unsat, that the user can then forward to the more powerful implies endpoint.

Just want to make sure that when we get this response back, what it actually means is _there is absolutely no overlap between the terms, we know because we found differing constructors_.